### PR TITLE
Collapsable TOC  - new PageBreak plugin options and improved view

### PR DESCRIPTION
**Pull request by LivioCavallo**
### The Problem
Titles and pagebreak chapter names in Articles tend to be quite long, particularly in european (continental) languages (English is shorter), so you end up having a huge TOC that occupies ALL THE PAGE WIDTH, extending for MANY ROWS...
In that situation the TOC occupies all the browser windows so:
- the Title is no more the most evident information in the page, as it is someway "shadowed" by that big TOC, and
- the article body overflows at the botton and you have to scroll down to read it.

### The Solution Proposed by this PR
If you could have a main TITLE, followed by an INDICATION that there are subsequent chapters, with a button to EVENTUALLY SHOW them (the TOC), followed by the article body (all possibly contained in the browser window), I think the layout will be cleaner and more readable.

### Summary of Changes
Added two options in PageBreak plugin so you can select if TOC will be always visible or it can be expanded/collapsed by user using a toggle button; you can set default toc status too (collapsed/expanded).
Minimal update in beez3 and protostar templates to accomodate padding and margins when toggle button is shown.

### Testing Instructions
Create and publish an Article with some pagebreaks, with long chapter names (as it's usual, at least in continental european languages).

### Expected Result
The Article TITLE should be evident and the article BODY first rows should be readable without having to scroll down the browser windows.

### Actual Result
The TOC occupies the main portion of the browser windows, so the Title is not the prominent information and the article body is not readable without scrolling down.

### Additional Info
In J!3.7.5 and in staging branch there is no option for article TOC to be user expandable/collapsable.

With this PR, article TOC will be collapsed/expanded on user click. To activate this feature you'll just have to go in pagebreak plugin and set: Collapsable TOC to Show  /  Default TOC show status to Hide

### CODE REVIEW

**Comment by brianteeman** [comment]
Changes to css should not be done directly in the css file but instead in the less files and then compiled using the generatecss.php script in the build folder - see https://docs.joomla.org/Joomla_LESS/en

**Comment by brianteeman** [comment]
Is there really a need for this?

**Comment by LivioCavallo** [comment]
Ok, thanks. I updated less file for protostar. Beez3 seems not to use .less to generate layout.css.

Far from beeing a philosophical question (what do we really need?) I understand your aim to keep Joomla! "clean", to avoid superflous functionalities and to include just really useful tools in Joomla!

For that goal I think that a simple tool like this, that allows the user to better organize the page layout and to save space will be useful.
In fact titles and pagebreak chapter names in user submitted articles tend to be quite long, particularly in european (continental) languages (English is shorter), so you end up having a huge TOC that occupies all the page width, extending for some rows... that occupies all the browser windows so the Title is no more the main information in the page, as it is someway "shadowed" by that big TOC, and the article body overflows at the botton and you have to scroll down to read it.
If you could have a main TITLE, followed by an indication that there are subsequent chapters, with a button to show them (the TOC), followed by the article body (all possibly contained in the browser window), I think the layout will be cleaner and more readable.

My 2c.

File: media/jui/less/navs.less [hunk 1]
```diff
@@ -117,6 +117,10 @@
 .nav-tabs > li {
   margin-bottom: -1px;
 }
+// Avoid last item in list to have border overlapped and hidden
+.nav-tabs > li:last-child {
+	margin-bottom: 0px;
```
Comment (mbabker) [inline media/jui/less/navs.less]: This is editing the Bootstrap source files.  Can this not be done at the template level?

Comment (LivioCavallo) [reply media/jui/less/navs.less]: Yes, it can.
Done.

**Comment by brianteeman** [comment]
Is there a reason that this new functionality is disabled if you dont display the title?

Is there a better way to display and describe the show/hide button - it just looks wrong

**Comment by LivioCavallo** [comment]
If the title is not shown, the user does not know what the button refers to, given the text "show/hide"; so I tought it was better to keep things togheter.
An alternative would be to use another text in button, something like "Toggle article index"; this would also permit to change code logic to show the button also if title is not displayed.
I am not sure what is better.

**Comment by brianteeman** [comment]
i still dont see the need for this in the core. it could easily be an alternative plugin

**Comment by ghost** [comment]
Status is set on "Needs Review".<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17660">issues.joomla.org/tracker/joomla-cms/17660</a>.</sub>

**Comment by LivioCavallo** [comment]
I updated the description to highlight that there is a problem, solved by this PR.

**Comment by Quy** [comment]
Rebase for J4?

**Comment by ghost** [comment]
@LivioCavallo please rebase on J4 as new Features go in J4.

Thanks for Hint @Quy.

**Pull request closed without merging**